spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestExpireSnapshotsProcedure.java

```diff
@@ -224,4 +236,60 @@ public void testConcurrentExpireSnapshotsWithInvalidInput() {
             catalogName, tableIdent, -1));
 
   }
+
+  @Test
+  public void testExpireDeleteFiles() throws Exception {
+    sql("CREATE TABLE %s (id bigint NOT NULL, data string) USING iceberg TBLPROPERTIES" +
+        "('format-version'='2', 'write.delete.mode'='merge-on-read')", tableName);
+
+    sql("INSERT INTO TABLE %s VALUES (1, 'a'), (2, 'b'), (3, 'c'), (4, 'd')", tableName);
```

Comment:
I am afraid we don't know the number of files that this insert will produce. That's why ID = 1 may end up in a separate file (unlikely but possible). I think it would be safer to use a typed `Dataset` and `SimpleRecord`. That way, we can call `coalesce(1)` before writing to make sure we produce only 1 file and a subsequent metadata delete is not possible. If we write just a single file with ID = 1, the DELETE operation below will be a metadata operation and the test will fail.